storage/range.go

```diff
@@ -1962,5 +1955,7 @@ func (r *Range) ChangeReplicas(changeType proto.ReplicaChangeType, replica proto
 	if err != nil {
 		return util.Errorf("change replicas of %d failed: %s", desc.RaftID, err)
 	}
+	// Enqueue this range again to see if there are more changes to be made.
+	r.rm.ReplicateQueue().MaybeAdd(r, r.rm.Clock().Now())
```

Comment:
splitQueue does multiple splits in one process() call. I didn't want to do that in replicateQueue.process because prioritization of replication actions may vary dramatically, so it's better to re-submit the range to be prioritized appropriately (although I'm not sure how big of a problem it would be in practice. Configurations with an even number of replicas are suboptimal so it may be better to fully update each range sequentially instead of bringing all the ranges to an even number before adding the next replica to any of them). We can't call MaybeAdd from process() because it will deadlock inside baseQueue.